modulegen/internal/vscode/types.go

```diff
@@ -1,34 +1,43 @@
 package vscode
 
-import "path/filepath"
+import (
+	"sort"
+)
 
-type ProjectDirectories struct {
-	RootDir            string
-	ModuleGeneratorDir string
-	Examples           []string
-	Modules            []string
+type Config struct {
+	Folders []Folder `json:"folders"`
 }
 
-func newProjectDirectories(rootDir string, examples []string, modules []string) *ProjectDirectories {
-	rootDirAbs, err := filepath.Abs(rootDir)
-	if err != nil {
-		rootDirAbs = rootDir
-	}
-
-	moduleGeneratorDirAbs := filepath.Join(rootDirAbs, "modulegen")
+type Folder struct {
+	Name string `json:"name"`
+	Path string `json:"path"`
+}
 
-	for i, example := range examples {
-		examples[i] = filepath.Join(rootDirAbs, "examples", example)
+func newConfig(examples []string, modules []string) *Config {
+	config := Config{
+		Folders: []Folder{
+			{
+				Path: "../",
+				Name: "testcontainers-go",
+			},
+			{
+				Path: "../modulegen",
+				Name: "modulegen",
+			},
+		},
 	}
-
-	for i, module := range modules {
-		modules[i] = filepath.Join(rootDirAbs, "modules", module)
+	for _, example := range examples {
+		config.Folders = append(config.Folders, Folder{
+			Path: "../examples/" + example,
+			Name: "examples/" + example,
+		})
 	}
-
-	return &ProjectDirectories{
-		RootDir:            rootDirAbs,
-		ModuleGeneratorDir: moduleGeneratorDirAbs,
-		Examples:           examples,
-		Modules:            modules,
+	for _, module := range modules {
+		config.Folders = append(config.Folders, Folder{
+			Path: "../modules/" + module,
+			Name: "modules/" + module,
+		})
 	}
+	sort.Slice(config.Folders, func(i, j int) bool { return config.Folders[i].Name < config.Folders[j].Name })
```

Comment:
You can replace Name by Path or define different names so they follow an order you like.

Reply:
In that case, let's have core library at the beginning. modulegen can be in its own place, but let's keep core with more visibility

Reply:
I'm  on it